applications/ssal-commods-dex.md

```diff
@@ -0,0 +1,239 @@
+
+### Team Code Repos
+Original Riso Blockchain https://github.com/MatteoPerona/Riso  
+Mock Frontend for Riso https://github.com/AlejandroSoumah/riso-front-end
+
+Matteo Perona https://github.com/MatteoPerona  
+Alejandro Soumah https://github.com/AlejandroSoumah  
+Taerim Choi https://github.com/cherrytoi 
+
+### Team LinkedIn Profiles (if available)
+Matteo Perona https://www.linkedin.com/in/matteo-perona/  
+Alejandro Soumah https://www.linkedin.com/in/alejandro-soumah/  
+Taerim Choi https://www.linkedin.com/in/taerim-choi/
+
+
+## Development Status :open_book:
+The following files, also attached above, are from the competition we participated in to test out our idea before looking for a grant. 
+
+Mock Frontend https://github.com/AlejandroSoumah/riso-front-end  
+Original Blockchain Concept https://github.com/MatteoPerona/Riso  
+Files from the competition https://github.com/MatteoPerona/riso-concept-files
+
+The last link attached contains our original concept white paper along with a one-pager and a pitch deck covering our original idea. As was mentioned before, we fleshed out this idea while working in a startup competition, so the materials are less technical and more business minded. The whitepaper was from an iteration of the app for which we built a rudimentary substrate based chain with a commodities pallet. In our next iteration we would like to take the functionality we built into that pallet and reduce it to a set of smart contracts. 
+
+Other resources we used while researching:  
+The Watr Whitepaper https://watr.org/wp-content/uploads/2023/06/Watr-Whitepaper-1.pdf  
+ADB Sustainable Development Working Paper Series https://www.adb.org/sites/default/files/publication/29972/adb-wp-25-commodities-exchange.pdf  
+
+## Development Roadmap :nut_and_bolt:
+### Overview
+- **Total Estimated Duration:** 3 Months
+- **Full-Time Equivalent (FTE):** 3
+- **Total Costs:** 30,000 USD
+
+### Milestone 1 Example — Basic Smart Contracts and UI
+
+- **Estimated duration:** 1 months
+- **FTE:**  2,3
+- **Costs:** 10,000 USD
+
+| Number | Deliverable | Specification |
+| -----: | ----------- | ------------- |
+| **0a.** | License | GPLv3 |
+| **0b.** | Documentation | We will provide both **inline documentation** of the code and a basic **tutorial** in our web documentation that explains how a user can interact with our smart contracts through CLI. |
+| **0c.** | Testing and Testing Guide | Core functions will be fully covered by comprehensive unit tests to ensure functionality and robustness. In the guide, we will describe how to run these tests. |
+| **0d.** | Docker | We will provide a Dockerfile(s) that can be used to test all the functionality delivered with this milestone. |
+| 1. | Smart Contracts | We will write the requisite ink! smart contracts to create, buy, and sell commodities contracts, testing them on Shibuya testnet. |
```

Comment:
Could you add more details about the design of the contracts? What data is required on-chain, which functions they will expose, how they interact, etc.